### PR TITLE
Migrate workflow to delete old Cloudsmith artifacts to GitHub Actions

### PR DESCRIPTION
**Pull request by glebashnik**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

## What

Adds a GitHub action workflow to delete old Cloudsmith artifacts.
The workflow is triggered either manually or on a daily schedule.

## Why

Part of the process of retiring Screwdriver as CI/CD runner.

## Additional Info

The corresponding Screwdriver pipeline is not removed since we don't have a dashboard for Github Actions yet.

See successful test run:
https://github.com/vespa-engine/vespa/actions/runs/10387039260/job/28759628746

### CODE REVIEW

**Comment by glebashnik** [comment]
@bjormel Could you look at this test run and verify that it deleted what it should and no more:
https://github.com/vespa-engine/vespa/actions/runs/10387039260/job/28759628746

I haven't changed the core logic so should be ok but just in case.

File: .github/scripts/delete-old-cloudsmith-artifacts.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+for VERSION in $VERSIONS_TO_DELETE; do
+  curl -sLf --header 'accept: application/json' \
+    "https://api.cloudsmith.io/v1/packages/vespa/open-source-rpms/?query=version:${VERSION}" | jq -re '.[] | .slug' >> $RPMS_TO_DELETE
+done
+
+echo "Deleting the following RPMs:"
+cat $RPMS_TO_DELETE
+
+if [ "$GITHUB_EVENT_NAME" == "pull_request" ]; then
```
Comment (esolitos) [inline .github/scripts/delete-old-cloudsmith-artifacts.sh:33]: This should actually be the other way around! It should **not** delete during a PR. 😅 

```suggestion
if [ "$GITHUB_EVENT_NAME" != "pull_request" ]; then
```

**Review by esolitos — CHANGES_REQUESTED** [review]
Found an actual bug there!

File: .github/scripts/delete-old-cloudsmith-artifacts.sh (hunk 1 above)
Comment (esolitos) [reply .github/scripts/delete-old-cloudsmith-artifacts.sh:33]: Alternatively:
```suggestion
if [ "$GITHUB_EVENT_NAME" == "schedule" ]; then
```

Comment (glebashnik) [reply .github/scripts/delete-old-cloudsmith-artifacts.sh:33]: Nice, thanks!

**Review by esolitos — APPROVED** [review]

**Pull request merged**